src/transformers/models/bark/modeling_bark.py

```diff
@@ -961,20 +961,28 @@ def generate(
         if attention_mask is not None:
             input_ids = input_ids.masked_fill((1 - attention_mask).bool(), semantic_generation_config.text_pad_token)
 
+        # ensure that the semantic prompts are all of the same length for batch processing
         if history_prompt is not None:
-            semantic_history = history_prompt["semantic_prompt"][-max_input_semantic_length:]
-            semantic_history = nn.functional.pad(
-                semantic_history,
-                (0, max_input_semantic_length - len(semantic_history)),
-                value=semantic_generation_config.semantic_pad_token,
-                mode="constant",
-            )
+            semantic_history = torch.cat(
+                [
+                    nn.functional.pad(
+                        sp[-max_input_semantic_length:],
+                        (0, max_input_semantic_length - len(sp[-max_input_semantic_length:])),
+                        value=semantic_generation_config.semantic_pad_token,
+                        mode="constant",
+                    ).unsqueeze(0)
+                    for sp in history_prompt["semantic_prompt"]
+                ],
+                dim=0,
+            ).to(self.device)
+
```

Comment:
After spending some time looking into this into more details, there is still the issue driven by the fact that the `semantic_output` is not the same length if I process 2 samples simultaneously instead of processing it separately.

Hence, the following:

```
voice_preset = "v2/fr_speaker_0"
inputs = processor("Je m'appelle Bob.", voice_preset=voice_preset)

model = BarkModel.from_pretrained("suno/bark")
audio_array = model.generate(**inputs, do_sample=False)
```

will lead to `semantic_output` smaller Than the tensor associated to the same prompt and voice preset in the following case:

```
processor = AutoProcessor.from_pretrained("suno/bark")

voice_preset = ["v2/fr_speaker_0", "v2/en_speaker_0"]

inputs = processor(["Je m'appelle Bob.", "Yo, bro, my name is bobby boy"], voice_preset=voice_preset)

model = BarkModel.from_pretrained("suno/bark")
audio_array = model.generate(**inputs, do_sample=False, temperature=1)
```

The outputs overall make sense and seem to match in both cases but this difference is still unwanted. 

A solution would be to remove the end padding for the `semantic_output` and do the processing as you mentioned and as I have currently implemented for `BarkCoarseModel`. And potentially do the same for the `coarse_output` in the case of `BarkFineModel` if necessary.

Reply:
Hey @MidAtBest, not sure to follow what's the issue here, can you expand on this ?